### PR TITLE
PROD-578: Fix box single click 

### PR DESCRIPTION
**Pull request by neel-ds**
- Description
In merge conflicts, create mysteryBox based on allowlist logic was removed and this PR reverts it. Additionally, it enables single click on box for opening and adjust box size once opened.

- What are the steps to test that this code is working?
You can open unclaimed mystery box prize on reveal tab and open it via box instead of button

- Screen shots or recordings for UI changes
NA

### CODE REVIEW

File: app/actions/claim.ts [hunk 1]
```diff
@@ -91,9 +91,9 @@ export async function claimAllAvailable() {
 
   if (!claimableQuestionIds.length) throw new Error("No claimable questions");
 
-  const FF_MYSTERY_BOX = process.env.NEXT_PUBLIC_FF_MYSTERY_BOX_CLAIM_ALL;
+  const isUserAllowed = await isUserInAllowlist();
 
-  const mysteryBoxId = FF_MYSTERY_BOX
+  const mysteryBoxId = isUserAllowed
```
Comment (samott) [inline app/actions/claim.ts:96]: I thought the allowlist was only for Chompmas?

Comment (neel-ds) [reply app/actions/claim.ts:96]: The allowlist was not limited to Chompmas. Its purpose is to replace the feature flag and provide initial access to the mystery box for a selective group of users, which cannot be achieved through FF alone.

File: app/actions/claim.ts [hunk 2]
```diff
@@ -91,9 +91,9 @@ export async function claimAllAvailable() {
 
   if (!claimableQuestionIds.length) throw new Error("No claimable questions");
 
-  const FF_MYSTERY_BOX = process.env.NEXT_PUBLIC_FF_MYSTERY_BOX_CLAIM_ALL;
+  const isUserAllowed = await isUserInAllowlist();
```
Comment (yash431garg) [inline app/actions/claim.ts:94]: Can you rename it to something like isEligibleForMysteryBox?

**Comment by neel-ds** [comment]
@yash431garg I am aware of it but in mobile screen it would compensate that space. It won't overlap with the above heading.

File: app/actions/claim.ts (hunk 2 above)
Comment (neel-ds) [reply app/actions/claim.ts:94]: yeah, will do it

**Comment by neel-ds** [comment]
I made the box size responsive according to screen using tailwind screen clamps.

**Review by yash431garg — DISMISSED** [review]
Look small in desktop but can be fixed in a follow up task.

**Comment by neel-ds** [comment]
> Look small in desktop but can be fixed in a follow up task.

Hope recent commit would do the needful
Screenshot 2024-12-23 at 8 11 28 PM

**Review by yash431garg — APPROVED** [review]

**Review by samott — APPROVED** [review]

**Pull request merged**